src/contract.rs

```diff
@@ -1,31 +1,40 @@
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::{Response, StdResult};
 use cw_storage_plus::Item;
+#[cfg(not(feature = "library"))]
+use sylvia::cw_std::Empty;
+use sylvia::cw_std::{Response, StdResult};
 
 use sylvia::contract;
-use sylvia::types::{ExecCtx, InstantiateCtx, QueryCtx};
+use sylvia::types::{CustomMsg, CustomQuery, ExecCtx, InstantiateCtx, QueryCtx};
 
-pub struct CounterContract {
+pub struct CounterContract<E, Q> {
     pub count: Item<u64>,
+    _phantom: std::marker::PhantomData<(E, Q)>,
 }
 
-#[cfg_attr(not(feature = "library"), sylvia::entry_points)]
+#[cfg_attr(not(feature = "library"), sylvia::entry_points(generics<Empty, Empty>))]
```

Comment:
It's smart to remove `cosmwasm_std` dependency and try to use `sylvia::cw_std`. For some reason though there's an error that appears in `sylvia::cw_std::entry_point` macro:
```
error[E0433]: failed to resolve: could not find `cosmwasm_std` in the list of imported crates
  --> src/contract.rs:15:38
   |
15 | #[cfg_attr(not(feature = "library"), sylvia::entry_points(generics<Empty, Empty>))]
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `cosmwasm_std` in the list of imported crates
   |
   = note: this error originates in the attribute macro `sylvia::cw_std::entry_point` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I'm afraid that `entry_point` macro from cosmwasm_std looks for `cosmwasm_std` that should be imported directly in the contract crate.

To reproduce this error type: `cargo wasm` in the generated contract crate.

Reply:
Good catch!
The `cosmwasm_std::entry_point` macro expands to:

```rust
mod __wasm_export_sudo {
    #[no_mangle]
    extern "C" fn sudo(ptr_0: u32, ptr_1: u32) -> u32 {
        ::cosmwasm_std::do_sudo(&super::sudo, ptr_0, ptr_1)
    }
}
```

It seems for now the `cosmwasm-std` dependency has to stay.

Reply:
I added `cargo wasm` and `cosmwasm-check` step in the CI to avoid this in the future. Thank you for this finding.